ospf6d/ospf6_gr_helper.c

```diff
@@ -155,7 +155,9 @@ static int ospf6_extract_grace_lsa_fields(struct ospf6_lsa *lsa,
 
 	lsah = (struct ospf6_lsa_header *)lsa->header;
 
-	length = ntohs(lsah->length) - OSPF6_LSA_HEADER_SIZE;
+	length = (ntohs(lsah->length) > OSPF6_LSA_HEADER_SIZE)
```

Comment:
Still think this isn't quite right - I'd rather not return SUCCESS, for example, since that may permit further processing in `ospf6_process_grace_lsa()`. How about something like this:


```
+       if (ntohs(lsah->length) <= OSPF6_LSA_HEADER_SIZE)
+               return OSPF6_FAILURE;
+
        length = ntohs(lsah->length) - OSPF6_LSA_HEADER_SIZE;
 
```

Reply:
Thanks! Done! @mjstapp